src/components/elements/LoadingButton.tsx

```diff
@@ -14,21 +14,26 @@ export interface LoadingButtonProps extends ButtonProps {
   loadingPosition: 'start' | 'end' | 'center';
 }
 
-const LoadingButton = forwardRef<LoadingButtonProps, any>((props, ref) => {
-  const extraSx = useMemo(() => {
-    if (props.loading && props.loadingPosition === 'start') {
-      return { pl: 5 };
-    }
-    if (props.loading && props.loadingPosition === 'end') {
-      return { pr: 5 };
-    }
-    return;
-  }, [props.loading, props.loadingPosition]);
+const LoadingButton = forwardRef<HTMLButtonElement, LoadingButtonProps>(
+  function LoadingButton(
+    props: LoadingButtonProps,
```

Comment:
I think adding some defaults here fix the type issues `{ loadingPosition = 'start', loading = false, ...props },`... sorry I completely messed up my `forwardRef`s causing all these type issues 🤦